tests/test_client.py

```diff
@@ -393,3 +393,33 @@ def test_client_errors(self):
         err = oqe.ConnectionError("oh no")
         assert err.args[0] == "oh no"
         assert err.err == "oh no"
+
+    @mock.patch("openqa_client.client.OpenQA_Client.openqa_request", autospec=True)
+    def test_get_latest_build(self, fakerequest):
+        client = oqc.OpenQA_Client()
+        with pytest.raises(TypeError):
+            client.get_latest_build()
+        fakerequest.return_value = {"build_results": [{"all_passed": 0, "build": "qq"}]}
+        ret = client.get_latest_build(42)
+        # returns default when no passed builds available
+        assert ret == "0"
+        ret = client.get_latest_build(42, all_passed=False)
+        # returns default when no passed builds available
+        assert ret == "0"
+        ret = client.get_latest_build(42, default_return="11")
+        # returns defined build when no builds were found
+        assert ret == "11"
+        fakerequest.return_value = {
+            "build_results": [
+                {"all_passed": 1, "build": "2"},
+                {"all_passed": 1, "build": "3"},
+                {"all_passed": 1, "build": "4"},
+                {"all_passed": 0, "build": "5"},
+            ]
+        }
+        ret = client.get_latest_build(42)
+        # returns lastest passed build when all_passed flag set to True
+        assert ret == "4"
+        ret = client.get_latest_build(42, all_passed=False)
+        # returns lastest failed build when all_passed flag set to False
+        assert ret == "5"
```

Comment:
we seem to be missing any testing of the custom sort key argument here.

Reply:
done

Reply:
thanks.